enumerables.rb

```diff
@@ -0,0 +1,67 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    length.times { |i| yield(self[i]) }
+    self
+  end
```

Comment:
- when a block is given when self is a range returns the range itself after calling the given block once for each element in self
- when a block is given when self is a hash behaves like #each calls the given block once for each element in self